dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/WmmAnalysis.java

```diff
@@ -28,7 +28,10 @@ public class WmmAnalysis {
             name= LOCALLY_CONSISTENT,
             description="Assumes local consistency for all created wmms.",
             secure=true)
-    private boolean assumeLocalConsistency = true;
+    // Set to false because local consistency does not hold in PTX
+    // When location accessed via different proxies but not properly synchronized,
+    // they can form intra-thread data races.
+    private boolean assumeLocalConsistency = false;
```

Comment:
I think we should make the default case somehow aware of the source-language and then choose `true` for non-PTX programs, and `false` for PTX.
I think it is possible to check with `config.hasProperty()` whether the user specified the local consistency assumption. If so, we take whatever the chose (even if it is wrong), and if not, we default to a source-language-aware default value.
`config.hasProperty` is deprecated, but I think this is a valid use-case here.
Alternatively we just override the user-specified value with `false` for PTX (i.e., we ignore the user input in that case).

Reply:
Updated